test/parse/test_entity_declaration.rb

```diff
@@ -23,6 +23,291 @@ def parse(internal_subset)
     end
 
     public
+
+    # https://www.w3.org/TR/2006/REC-xml11-20060816/#NT-GEDecl
+    class TestGeneralEntityDeclaration < self
+      # https://www.w3.org/TR/2006/REC-xml11-20060816/#NT-Name
+      class TestName < self
+        def test_prohibited_character
+          exception = assert_raise(REXML::ParseException) do
+            REXML::Document.new('<!DOCTYPE root [<!ENTITY invalid&name "valid-entity-value">]>')
+          end
+          assert_equal(<<-DETAIL.chomp, exception.to_s)
+Malformed entity declaration
+Line: 1
+Position: 61
+Last 80 unconsumed characters:
+ invalid&name \"valid-entity-value\">]>
+          DETAIL
+        end
+      end
+
+      # https://www.w3.org/TR/2006/REC-xml11-20060816/#NT-EntityDef
+      class TestEntityDefinition < self
+        # https://www.w3.org/TR/2006/REC-xml11-20060816/#NT-EntityValue
+        class TestEntityValue < self
+          def test_no_quote
+            exception = assert_raise(REXML::ParseException) do
+              REXML::Document.new('<!DOCTYPE root [<!ENTITY valid-name invalid-entity-value>]>')
+            end
+            assert_equal(<<-DETAIL.chomp, exception.to_s)
+Malformed entity declaration
+Line: 1
+Position: 59
+Last 80 unconsumed characters:
+ valid-name invalid-entity-value>]>
+            DETAIL
+          end
+
+          def test_prohibited_character
+            exception = assert_raise(REXML::ParseException) do
+              REXML::Document.new('<!DOCTYPE root [<!ENTITY valid-name "% &">]>')
+            end
+            assert_equal(<<-DETAIL.chomp, exception.to_s)
+Malformed entity declaration
+Line: 1
+Position: 44
+Last 80 unconsumed characters:
+ valid-name \"% &\">]>
+            DETAIL
+          end
+
+          def test_unnecessary_ndata_declaration
```

Comment:
Could you move this to `TestEntityDefinition` and use `test_entity_value_and_notation_data_declaration`?

This test case is for `EntityValue` not `EntityDef`.

Reply:
Fixed at https://github.com/ruby/rexml/pull/183/commits/c211ab51ff7af2a17f1a276f637b3b9e215ad488